hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java

```diff
@@ -817,14 +817,18 @@ public LockOptions getLockOptions() {
 
 	@Override
 	public SqmQueryImplementor<R> setLockOptions(LockOptions lockOptions) {
-		verifySelect();
+		if ( lockOptions.getLockMode() != LockMode.NONE ) {
+			verifySelect();
+		}
 		getQueryOptions().getLockOptions().overlay( lockOptions );
 		return this;
 	}
 
 	@Override
 	public SqmQueryImplementor<R> setLockMode(String alias, LockMode lockMode) {
-		verifySelect();
+		if ( lockMode != LockMode.NONE ) {
+			verifySelect();
+		}
```

Comment:
IMO we should remove this restriction, as that is a Hibernate method and we actually do support locking in subqueries. Wdyt @dreab8, @sebersole ?

Reply:
I'm fine with removing this check here

Reply:
I made the changes to be in line with another method in the same class:
```
	@Override
	public SqmQueryImplementor<R> setLockMode(LockModeType lockMode) {
		if ( lockMode != LockModeType.NONE ) {
			verifySelect();
		}
		getSession().checkOpen( false );
		getQueryOptions().getLockOptions().setLockMode( LockMode.fromJpaLockMode( lockMode ) );
		return this;
	}
```
@sebersole , saw the code was written recently. Should we remove this check (maybe everywhere in this class)?